File: task3/task3a_b/src/main/java/com/task3a_b/Aircraft.java
package com.task3a_b;

import java.util.ArrayList;

public class Aircraft extends Transport{
    ArrayList<String> airSpaceList = new ArrayList<String>();
    private String airSpace;

    public Aircraft(){

    }

    public Aircraft(int id, String type, double speed, double cost){
        super(id, type, speed, cost);
    }

    @Override
    public void display_vehicle_details(){
        super.display_vehicle_details();
        System.out.println("Air Space List:");
        for (int i = 0; i < airSpaceList.size(); i++){
            System.out.println(airSpaceList.get(i));
        }
    }

    // Setters
    public void setAirSpace(String airSpace){
        this.airSpace = airSpace;
    }

    public void addToList(String airSpace){
        airSpaceList.add(airSpace);
    }

    public void updateAirSpaceinList(String airSpace, int pos){
        airSpaceList.set(pos, airSpace);
    }

    // Getters
    public String getRoute(int pos){
        String airSpaceAtPos = airSpaceList.get(pos);
        return airSpaceAtPos;
    }
}

File: task1/task1b/Aircraft.cpp
#include "Aircraft.h"
#include <iostream>
#include <string.h>

Aircraft::Aircraft(){}

Aircraft::Aircraft(int id, const string& type, double speed, double cost): Transport(id, type, speed, cost) {}

void Aircraft::display_transport_details() {
    Transport::display_transport_details();
    cout << "Air Space List: " << endl;
    for(int i = 0; i < airSpaceList.size(); i++){
        cout << airSpaceList.at(i) << endl;
    }
}

void Aircraft::updateAirSpaceinList(const string& airSpace, int pos){
    airSpaceList.at(pos) = airSpace;
}


// Setters
void Aircraft::setAirSpace(const string& airSpace){
    this->airSpace = airSpace;
}

void Aircraft::addToList(const string& airSpace){
    airSpaceList.push_back(airSpace);
}


// Getters
string Aircraft::getRoute(int pos) const{
    string airSpaceAtPos = airSpaceList.at(pos);
    return airSpaceAtPos;
}